pkg/controller/pxc/status.go

```diff
@@ -235,14 +147,31 @@ func (r *ReconcilePerconaXtraDBCluster) updateStatus(cr *api.PerconaXtraDBCluste
 		}
 		cr.Status.Status = api.AppStateError
 	case cr.Status.PXC.Status == api.AppStateInit ||
-		(cr.Spec.ProxySQL != nil && cr.Status.ProxySQL.Status == api.AppStateInit) ||
-		(cr.Spec.HAProxy != nil && cr.Status.HAProxy.Status == api.AppStateInit):
+		(cr.ProxySQLEnabled() && cr.Status.ProxySQL.Status == api.AppStateInit) ||
+		(cr.HAProxyEnabled() && cr.Status.HAProxy.Status == api.AppStateInit):
 		clusterCondition = api.ClusterCondition{
 			Status:             api.ConditionTrue,
 			Type:               api.ClusterInit,
 			LastTransitionTime: metav1.NewTime(time.Now()),
 		}
 		cr.Status.Status = api.AppStateInit
+	case (cr.Status.PXC.Status == cr.Status.ProxySQL.Status && cr.ProxySQLEnabled()) ||
+		(cr.Status.PXC.Status == cr.Status.HAProxy.Status && cr.HAProxyEnabled()):
+		if cr.Status.PXC.Status == api.AppStateReady {
+			clusterCondition = api.ClusterCondition{
+				Status:             api.ConditionTrue,
+				Type:               api.ClusterReady,
+				LastTransitionTime: metav1.NewTime(time.Now()),
+			}
+		}
+		cr.Status.Status = cr.Status.PXC.Status
+	case !cr.ProxySQLEnabled() && !cr.HAProxyEnabled() && cr.Status.PXC.Status == api.AppStateReady:
+		clusterCondition = api.ClusterCondition{
+			Status:             api.ConditionTrue,
+			Type:               api.ClusterReady,
+			LastTransitionTime: metav1.NewTime(time.Now()),
+		}
+		cr.Status.Status = cr.Status.PXC.Status
```

Comment:
This looks confusing and not easy to comprehend.
In fact, both of these cases produce status `Ready`. There is no other way - Init and Error are checked before. Everything else is "undefined behaviour" and [should be] covered in `default`.

Reply:
How about moving this switch to some `cr.Status.ClusterStatus()` method which would return the status?

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/866/commits/44519bab30d5df0b42a5b11a2accc27eb9290893